Functions/External/xASL_ext_VABYRun.m

```diff
@@ -0,0 +1,90 @@
+function [x, Result1] = xASL_ext_VABYRun(VABYCommand, x, NicenessValue, bVerbose)
+%xASL_ext_VABYRun Run VABY from Matlab (ExploreASL)
+%
+% FORMAT: [x] = xASL_adm_RunVABY(VABYCommand, x[, NicenessValue, bVerbose])
+%
+% INPUT:
+%   VABYCommand     - Command line job for VABY (REQUIRED)
+%   x               - structure containing fields with all information required to run this submodule (REQUIRED)
+%   NicenessValue   - the linux nice parameter, a scale with 40 integers 
+%                     as index of the priority granted to a process. Lower
+%                     = higher priority, higher = lower priority. If no
+%                     other processes are running, a lower priority will
+%                     still use many resources.
+%                     Provide a number between [-20 +19], (OPTIONAL,
+%                     DEFAULT=10)
+%   bVerbose        - verbose output (OPTIONAL, DEFAULT true)
+%
+% OUTPUT:
+%   x               - as input, outputting VABY dir (OPTIONAL)
+%   Result1         - Result1 describes if the execution was successful
+%                     (0 = successful, NaN = no VABY found, 1 or other = something failed)
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: This function runs an VABY command from ExploreASL:
+%
+%              1. Checking the VABY dir
+%              2. Running the command
+%
+% Supports .nii & .nii.gz, Linux, MacOS & Windows
+% 
+% EXAMPLE: xASL_ext_VABYRun(VABYCommand, x);
+% __________________________________
+% Copyright (C) 2015-2024 ExploreASL
+% Licensed under Apache 2.0, see permissions and limitations at
+% https://github.com/ExploreASL/ExploreASL/blob/main/LICENSE
+% you may only use this file in compliance with the License.
+% __________________________________
+
+
+
+%% Admin
+if nargin<2 || isempty(x)
+    warning('x input missing');
+    x = struct;
+end
+if nargin<3 || isempty(NicenessValue)
+    NicenessValue = 10;
+end
+if nargin<4 || isempty(bVerbose)
+    bVerbose = true;
+end
+
+% Defaults
+Result1 = NaN;
+
+%% Find VABY directory
+[VABYdir, x] = xASL_ext_VABYSetDir(x);
+
+if min(isnan(VABYdir))
```

Comment:
xASL_ext_VABYSetDir outputs NaN in case a valid path doesn't exist. But don't ask _me_ why, because you programmed it like that :) I would have preferred an empty string.

We have to use min(isnan(VABYdir)) or sum(isnan(VABYDIR)) to make sure it works for both strings/chars/ and numbers(because NaN) is a number.
Contains(VABYdir, NaN) will not work because contains requires string on input and can't have NaNs as patterns.

Going back to the original comment -> I would never use NaNs as output because then you really mix chars and numbers and this creates issues.

Reply:
We always use NaN as a number, never as a string or char. I know I programmed this, I'm fine using [] instead. We just need consensus on when to use [] and when NaN. One is e.g. missing data and the other is incorrect code/processing.

I now understand your point, NaN is a number and the desired output is a char. OK, then we should change it to either 'n/a' (the BIDS equivalent of NaN) or just and empty char.

You can resolve for now

Reply:
Yes - it's the mixing of types that makes it tricky.

OK - I'm changing to []. Because this is function output not directly related to BIDS and it doesn't get out to the user - so nobody sees that N/A. 

You can resolve, I just kept it here to show that I indeed changed it to an empty char already.